samples/csharp/getting-started/Ranking_PersonalizedSort/PersonalizedRanking/Program.cs

```diff
@@ -0,0 +1,189 @@
+﻿using Microsoft.ML;
+using Microsoft.ML.Trainers.LightGbm;
+using PersonalizedRanking.Common;
+using PersonalizedRanking.DataStructures;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using static Microsoft.ML.DataOperationsCatalog;
+
+namespace PersonalizedRanking
+{
+    class Program
+    {
+        const string AssetsPath = @"../../../Assets";
+        readonly static string TrainDatasetPath = Path.Combine(AssetsPath, "InputData_Train.csv");
+        readonly static string TestDatasetPath = Path.Combine(AssetsPath, "InputData_Test.csv");
+        readonly static string ModelPath = Path.Combine(AssetsPath, "RankingModel.csv");
+
+        readonly static string OriginalDatasetPath = Path.Combine(AssetsPath, "Train.csv");
+        readonly static string OriginalExampleDatasetPath = Path.Combine(AssetsPath, "Test.csv");
+
+        static void Main(string[] args)
+        {
+            // Create a common ML.NET context.
+            // Seed set to any number so you have a deterministic environment for repeateable results.
+            MLContext mlContext = new MLContext(seed: 0);
+
+            try
+            {
+                PrepDatasets(mlContext, AssetsPath, OriginalDatasetPath, TrainDatasetPath, TestDatasetPath);
+
+                var model = TrainModel(mlContext, TrainDatasetPath, ModelPath);
+
+                EvaluateModel(mlContext, model, TestDatasetPath);
+
+                ConsumeModel(mlContext, ModelPath, OriginalExampleDatasetPath);
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e.Message);
+            }
+
+            Console.ReadLine();
+        }
+
+        static void PrepDatasets(MLContext mlContext, string assetPath, string originalDatasetPath, string trainDatasetPath, string testDatasetPath)
+        {
+            const string DatasetUrl = "https://www.kaggle.com/c/expedia-personalized-sort/download/data.zip";
```

Comment:
File is not public. Is requiring a user to sign up for a kaggle account too onerous? 

I'd recommend using the MSLR-WEB10K (or WEB30K) dataset from our CDN:
* https://aka.ms/mlnet-resources/benchmarks/MSLRWeb10KTrain720kRows.tsv
* https://aka.ms/mlnet-resources/benchmarks/MSLRWeb10KValidate240kRows.tsv
* https://aka.ms/mlnet-resources/benchmarks/MSLRWeb10KTest240kRows.tsv

As zip:
* MSLR-WEB10K: <br>https://express-tlcresources.azureedge.net/datasets/MSLR-WEB10K/MSLR-WEB10K.zip ([README](https://express-tlcresources.azureedge.net/datasets/MSLR-WEB10K/README.md), [LICENSE](https://express-tlcresources.azureedge.net/datasets/MSLR-WEB10K/LICENSE.md))
* MSLR-WEB30K: <br>https://express-tlcresources.azureedge.net/datasets/MSLR-WEB30K/MSLR-WEB30K.zip ([README](https://express-tlcresources.azureedge.net/datasets/MSLR-WEB30K/README.md), [LICENSE](https://express-tlcresources.azureedge.net/datasets/MSLR-WEB30K/LICENSE.md))

We use the MSLR-WEB10K dataset for our [ranking benchmarks](https://github.com/dotnet/machinelearning/blob/08318656ed0f0649aa75370b019cf4bcbda5a6a5/test/Microsoft.ML.Benchmarks/Numeric/Ranking.cs). The dataset is auto-downloaded from the above CDN aka.ms URLs when you run `build.cmd -- /t:DownloadExternalTestFiles /p:IncludeBenchmarkData=true` to [download/run](https://github.com/dotnet/machinelearning/blob/8444a5a2a4c6b162c35b98cbff95983db3d676c2/test/Microsoft.ML.Benchmarks/README.md#run-the-performance-tests) the benchmark datasets.

Reply:
Yes you do need a Kaggle account for this - I had assumed that all Kaggle datasets required you have an account.  But, based on your comment, that must not be the case.  You probably noticed but I have sent an email to CELA to learn if there is a possibility of using this dataset.  If it turns out I can't, then I will look at using the benchmark dataset.  Overall, I feel that the hotel data is a very compelling scenario which is why I decided to try pursuing it.

Reply:
I'd recommend all demos have datasets that can be downloaded automatically (no user action needed). Will CELA let us host the dataset in our CDN?

Granted Kaggle is pretty awesome and folks should hone there data science skill there. I think forcing our users to sign up for an external account before trying demo code is a bit rough.

We will have more ranking samples of course. For instance a short sample that teaches less aspects but is easy to modify for towards their own datasets. And perhaps a sample which does string comparisons (query vs. documents) for feature engineering.

Reply:
I have now switched the dataset to use the one you recommended for ranking search engine results.

#resolved